tools/reactive-controllers/src/ColorController.ts

```diff
@@ -0,0 +1,520 @@
+/*
+Copyright 2022 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import type { ReactiveElement } from 'lit';
+import Color from 'colorjs.io';
+import type {
+    ColorObject,
+    ColorTypes as DefaultColorTypes,
+} from 'colorjs.io/types/src/color';
+import type ColorSpace from 'colorjs.io/types/src/space';
+
+type ColorTypes =
+    | DefaultColorTypes
+    | {
+          r: number | string;
+          g: number | string;
+          b: number | string;
+          a?: number | string;
+      }
+    | {
+          h: number | string;
+          s: number | string;
+          l: number | string;
+          a?: number | string;
+      }
+    | {
+          h: number | string;
+          s: number | string;
+          v: number | string;
+          a?: number | string;
+      };
+
+export type { Color, ColorTypes };
+type ColorValidationResult = {
+    spaceId: string | null;
+    coords: number[];
+    isValid: boolean;
+    alpha: number;
+};
+export const extractHueSaturationValueAndAlphaRegExp =
+    /^hsva?\s?\((\d{1,3}\.?\d*?)%?,?\s?(\d{1,3})%?,?\s?(\d{1,3})%?,?\s?(\d\.?\d?)?/;
+
+export class ColorController {
```

Comment:
Don't for get to document this controller at `tools/reactive-controllers/color-controller.md` and link to it from `tools/reactive-controllers/README.md`.

Reply:
Done! and thanks for pointing this out